### PR TITLE
Implement Adversarial Patch attack in PyTorch.

### LINKED ISSUE #851: Implement Adversarial Patch framework-specific in PyTorch
We would like to implement Adversarial Patch framework-specific in PyTorch. This will facilitate future extensions for improved real world patches and improve performance and accuracy.

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

This pull request implements Adversarial Patch attack in PyTorch.

Fixes #851

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/evasion/adversarial_patch/adversarial_patch_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+    ):
+        """
+        Create an instance of the :class:`.AdversarialPatchPyTorch`.
+
+        :param classifier: A trained classifier.
+        :param rotation_max: The maximum rotation applied to random patches. The value is expected to be in the
+               range `[0, 180]`.
+        :param scale_min: The minimum scaling applied to random patches. The value should be in the range `[0, 1]`,
+               but less than `scale_max`.
+        :param scale_max: The maximum scaling applied to random patches. The value should be in the range `[0, 1]`, but
```
Comment (minhitbk) [inline art/attacks/evasion/adversarial_patch/adversarial_patch_pytorch.py:88]: `scale_min.` --> `scale_min`.

File: art/estimators/classification/pytorch.py [hunk 2]
```diff
@@ -301,6 +301,28 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> np.ndarray:
 
         return predictions
 
+    def _predict_framework(self, x, **kwargs):
+        """
+        Perform prediction for a batch of inputs.
+
+        :param x: Test set.
+        :type x: `np.ndarray`
```
Comment (minhitbk) [inline art/estimators/classification/pytorch.py:309]: Is `x` of tensor type?

File: art/estimators/classification/pytorch.py [hunk 3]
```diff
@@ -301,6 +301,28 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> np.ndarray:
...
+        :param batch_size: Size of batches.
```
Comment (minhitbk) [inline art/estimators/classification/pytorch.py:310]: I think `batch_size` is not necessary here.

File: art/estimators/classification/pytorch.py [hunk 4]
```diff
@@ -301,6 +301,28 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> np.ndarray:
...
+        :type batch_size: `int`
+        :return: Array of predictions of shape `(nb_inputs, nb_classes)`.
+        :rtype: `np.ndarray`
```
Comment (minhitbk) [inline art/estimators/classification/pytorch.py:313]: Return type is tensor?

**Review by minhitbk — APPROVED** [review]
Code looks good. Just a few minor comments.

File: art/attacks/evasion/adversarial_patch/adversarial_patch_pytorch.py (hunk 1 above)
Comment (beat-buesser) [reply art/attacks/evasion/adversarial_patch/adversarial_patch_pytorch.py:88]: Ok.

File: art/estimators/classification/pytorch.py (hunk 4 above)
Comment (beat-buesser) [reply art/estimators/classification/pytorch.py]: Ok.

File: art/estimators/classification/pytorch.py (hunk 3 above)
Comment (beat-buesser) [reply art/estimators/classification/pytorch.py]: Ok.

File: art/estimators/classification/pytorch.py (hunk 2 above)
Comment (beat-buesser) [reply art/estimators/classification/pytorch.py]: Ok.

**Comment by beat-buesser** [comment]
Hi @minhitbk Thank you very much for your review! I have addressed the comments above.
